### ISSUE TITLE
Map record in AC associates with wrong track

### ISSUE THREAD

**Issue by neilryoung** [zegreatclan/SimHub#303]
In Assetto Corsa the map record file created from driving the modern version of Spa circuit saves as Spa_1966. As this is a Kunos track I have assumed (always dangerous) that the problem is not caused by incorrect data for the track.

**Comment by zegreatclan** [comment]
Hi ! I will check ;) Could you give me some additional informations however ? was it online/offline, in a race ? And could you tell me which version of simhub you are using ?

**Comment by neilryoung** [comment]
I'm using version 6.5.1. but it was doing this on the earlier versions too.
It creates the spa_1966.tm.gz file regardless of what type of event (online or offline), it also associates the map file back with the track when you race there again (regardless of event type). It becomes a problem because there is a 1966 version of the track with a much longer route which then accesses the same file.
Where does Simhub find the track name which it uses to name the file?

Thanks, Neil.

**Comment by zegreatclan** [comment]
Could you make sure to update to the latest version first ? I remember having fixed something related to this a few versions ago. Currently track name comes both from API or ac config files, depending of the context  since AC API has a few bugs (online or when track name is too long)

**Comment by neilryoung** [comment]
Apologies I thought that I had already updated to the latest version but I had only downloaded iit without installing.
Updating to 6.6.4 has changed the issue, but fundamentally it still has the same problem.
The filename is now spa.tm.gz, but the 1966 version of the track is still trying to share this file with the modern Kunos track layout. 
One thing that I have noticed is that all other kunos tracks are prefixed with 'ks_'. For example,  'ks_monza66-full.tm.gz'.
Maybe Kunos have not stuck to a standard form with their Spa track data.

**Comment by zegreatclan** [comment]
Hi, i've just checked but spa 1966 is a mod, could you give me your download link, or eventually send it in a zip so i can test with your version of the track ?

**Comment by neilryoung** [comment]
Yes it is a mod track. It is from [https://actrackrebootproject.wixsite.com/ac-track-re-boot](url)
The track files are 193MB so I can't attached them to this post.
Thanks.

**Comment by zegreatclan** [comment]
Thanks ! Got it indeed, it was one of the workaround for bad telemetry data or truncated making it choose the wrong track. it will be fixed for nextion version

**Comment by neilryoung** [comment]
Awesome, thank you.

**Comment by neilryoung** [comment]
Thank you. This works perfectly for me now.
